src/azure-cli/requirements.py3.Darwin.txt

```diff
@@ -127,7 +127,7 @@ PyNaCl==1.5.0
 pyOpenSSL==22.1.0
 python-dateutil==2.8.0
 requests-oauthlib==1.2.0
-requests[socks]==2.26.0
+requests[socks]==2.31.0
```

Comment:
Maybe bump `urllib3` as well? `requests` and `urllib3` are usually bumped together (https://github.com/Azure/azure-cli/pull/18213, https://github.com/Azure/azure-cli/pull/19801).

Reply:
Here is the fix commit: https://github.com/psf/requests/commit/74ea7cf7a6a27a4eeb2ae24e162bcc942a6706d5
It looks like a logic error in requests.